batch_seqr_loader/batch_workflow.py

```diff
@@ -1310,15 +1459,20 @@ def _samples_to_add_to_db(
     tmp_bucket: str,
     gvcf_by_sid: Dict[str, str],
 ) -> Tuple[Set[str], Set[str], Set[str], bool, str]:
-    if utils.file_exists(genomicsdb_gcs_path):
-        # Check if sample exists in the DB already
+    if utils.file_exists(join(genomicsdb_gcs_path, 'callset.json')):
+        # Checking if samples exists in the DB already
         genomicsdb_metadata = join(local_tmp_dir, f'callset-{interval_idx}.json')
-        # This command will download the DB metadata file locally.
-        # The `-O` argument to `tar` means "write the file being extracted to the stdout",
-        # and the file to be extracted is specified as a positional argument to `tar`.
+        # The `-o GSUtil:check_hashes=never` flag is required to get around the gsutil
+        # integrity checking error, as conda gsutil doesn't use CRC32c:
+        # > Downloading this composite object requires integrity checking with CRC32c,
+        #   but your crcmod installation isn't using the module's C extension, so the
+        #   hash computation will likely throttle download performance.
+        #
+        #   To download regardless of crcmod performance or to skip slow integrity
+        #   checks, see the "check_hashes" option in your boto config file.
         cmd = (
-            f'gsutil cat {genomicsdb_gcs_path} | '
-            f'tar -O --extract workspace/callset.json > {genomicsdb_metadata}'
+            f'gsutil -o GSUtil:check_hashes=never cp '
+            f'{join(genomicsdb_gcs_path, "callset.json")} {genomicsdb_metadata}'
         )
         logger.info(cmd)
         subprocess.run(cmd, check=False, shell=True)
```

Comment:
I personally prefer subprocess, because it's more transparent, easier to debug and reproduce, and easier to transfer shell commands into python. But if you think it would be better to follow one standard in the group, happy to switch to a library :)

Reply:
But I see that such a technical detail as this long comment about the gsutil usage shouldn't be in a function called `_samples_to_add_to_db`. Just wrapped that stuff into a util function.